src/Orleans.Core/Async/UnobservedExceptionsHandler.cs

```diff
@@ -6,57 +6,33 @@ namespace Orleans
 {
     using Orleans.Runtime;
 
-    internal class UnobservedExceptionsHandlerClass
+    internal class UnobservedExceptionsHandler : IDisposable
     {
-        private static readonly Object lockObject = new Object();
-        private static ILogger logger;
-        private static UnobservedExceptionDelegate unobservedExceptionHandler;
-        private static readonly bool alreadySubscribedToTplEvent = false;
+        private readonly ILogger logger;
+        private UnobservedExceptionDelegate unobservedExceptionHandler;
 
         internal delegate void UnobservedExceptionDelegate(ISchedulingContext context, Exception exception);
         
-        static UnobservedExceptionsHandlerClass()
+        public UnobservedExceptionsHandler(ILogger<UnobservedExceptionsHandler> logger)
         {
-            lock (lockObject)
-            {
-                if (!alreadySubscribedToTplEvent)
-                {
-                    TaskScheduler.UnobservedTaskException += InternalUnobservedTaskExceptionHandler;
-                    alreadySubscribedToTplEvent = true;
-                }
-            }
+            this.logger = logger;
+            TaskScheduler.UnobservedTaskException += InternalUnobservedTaskExceptionHandler;
         }
 
-        internal static void InitLogger(ILoggerFactory loggerFactory)
-        {
-            logger = loggerFactory.CreateLogger<UnobservedExceptionsHandlerClass>();
-        }
 
-        internal static bool TrySetUnobservedExceptionHandler(UnobservedExceptionDelegate handler)
+        internal bool TrySetUnobservedExceptionHandler(UnobservedExceptionDelegate handler)
```

Comment:
Since this class is now only usable by the silo, please avoid this 2-step initialization, and just have the handler log directly, instead of requiring the silo to pass it how to log.
Also, can now move to Orleans.Runtime project